### PR TITLE
Generate contents table for API

### PR DESCRIPTION
**Pull request by SemenchenkoVitaliy**
Closes: https://github.com/metarhia/metadoc/issues/74

### CODE REVIEW

File: .metadocrc [hunk 1]
```diff
@@ -1,6 +1,8 @@
 {
   "headerFile": "doc/header.md",
   "footerFile": "doc/footer.md",
+  "generateContentsTable": true,
+  "removeInterface": true,
```
Comment (lundibundi) [inline .metadocrc]:
```suggestion
```
?

File: README.md [hunk 2]
```diff
@@ -28,7 +28,9 @@ Options:
 
 ## API
 
-### Interface: doc
+- [introspect](#introspectnamespace-text)
```
Comment (lundibundi) [inline README.md]: I'd prefer to caption this somehow. (i.e. `Table of Contents`)

File: bin/cli.js [hunk 3]
```diff
@@ -30,6 +30,10 @@ const args = yargs
     type: 'boolean',
     describe: 'Generate documentation for methods that have #private notation',
   })
+  .option('generate-contents-table', {
```
Comment (lundibundi) [inline bin/cli.js]: I'd prefer to name this `contents` in the spirit of `private` option name.

File: test/example.md [hunk 4]
```diff
@@ -1,3 +1,39 @@
+- [Interface example](#interface-example)
```
Comment (lundibundi) [inline test/example.md]: Can we somehow limit the level of entries that get pulled into the Contents? (i.e. say only 1st and 2nd level entries are included in the contents)
I think we can add this as an optional parameter to `--contents` option.

File: bin/cli.js (hunk 3 above)
Comment (SemenchenkoVitaliy) [reply bin/cli.js]: Maybe `contents-table`?

File: .metadocrc (hunk 1 above)
Comment (SemenchenkoVitaliy) [reply .metadocrc]: Since there is only one interface, I'd prefer to not generate the header for it and IMO it looks better without interface header.

Comment (lundibundi) [reply .metadocrc]: I agree, but this at least shouldn't be a part of the same commit.

**Comment by lundibundi** [comment]
Landed in cc8b7a3c01319971c02306d4aacf328aa737512c, 039bf3f3f3e6fef9853439ec643ee396a2796c59.

**Pull request closed without merging**